boards/mips/pic32mz/chipkit-wifire/README.txt

```diff
@@ -0,0 +1,60 @@
+README
+======
+
+  chipKIT Wi-FIRE board
+  (https://chipkit.net/wiki/index.php?title=ChipKIT_Wi-Fire) features
+  the Microchip PIC32MZ2048EFG100 MCU, Microchip MRF24WG0MA Wi-Fi module,
+  USB 2.0 Full-Speed / Hi-Speed OTG controller, Micro SD card connector etc.
+  The PIC32MZ2048EFG100 is a 200 MHz 32-bit MIPS M5150 operation with 2MB Flash
+  memory and 512KB RAM.
+
+Contents
+========
+
+  - GCC toolchain installation
+  - Building NuttX for the chipKIT Wi-FIRE board
+  - Programming Flash
+  - Serial console
+
+GCC toolchain installation
+==========================
+
+  $ wget https://sourcery.mentor.com/GNUToolchain/package12725/public/mips-sde-elf/mips-2014.05-24-mips-sde-elf-i686-pc-linux-gnu.tar.bz2
+  $ sudo tar xvjf mips-2014.05-24-mips-sde-elf-i686-pc-linux-gnu.tar.bz2 -C /usr/local
+
+Building NuttX for the chipKIT Wi-FIRE board
+============================================
+
+  $ cd nuttx
+  $ make apps_distclean
+  $ make distclean
+  $ ./tools/configure.sh -l chipkit-wifire:nsh
+  $ make CROSSDEV=mips-sde-elf-
+
+Programming Flash
+=================
+
+  CAUTION:
+
+  Currently the pre-installed bootloader of the chipKIT Wi-FIRE board
+  is not supported in this NuttX configuration. Doing the following
+  steps will erase the factory installed bootloader in the Flash
+  memory! If you are going to recover the bootloader later on your
+  chipKIT Wi-FIRE board, the original chipKIT bootloader can be found here:
+  https://reference.digilentinc.com/_media/chipkit_wifire/chipkit-wifire-v01000303.zip
```

Comment:
Can't we reserve the bootloader region in the linker script?

Reply:
Can you share the full idea?

Reply:
I think just reserving a bootloader region in the linker script would be insufficient.  You would also need to capture the power on reset exception in the NuttX code.  If you can get the reset, then you can set the VBAR to redirect the other exceptions.

Reply:
We can't reserve the bootloader region and then flash NuttX using the bootloader?  If it's too tailored to an Arduino environment that could be impossible (or requires some twerking to the binary to make it compatible.)

In any case, if NuttX can start without the bootloader than it won't really hurt to overwrite it.